### ISSUE TITLE
[error] [gradle-server] Gradle version is required

### ISSUE THREAD

**Issue by Eskibear** [microsoft/vscode-gradle#922]
Extension Name: vscode-gradle
Extension Version: Private build based on PR #919 
OS Version: Windows 10
VSCode version: 1.59.0

**Describe the bug**
Nothing is working, except a message "Gradle: Build Error" on status bar. 

**To Reproduce**
1. use spring initializr to create a simple gradle project.
2. open it, and you see the screenshot

**Expected behavior**
1. I expect it's working out-of-box, but I don't see much insightful message from the log after I click the status bar item.
2. the "Gradle: Build Error" is not obvious, so I spend time wondering what's happening. Maybe a warning/error color is better?
3. the "Gradle: Build Error" stays there forever. How do I get it resolved?

**Screenshots**
![image](https://user-images.githubusercontent.com/2351748/129997108-ea193d8c-4058-4c60-871d-9dc0ef03a891.png)

**Output from "Gradle Tasks"**
```
[info] [gradle-server] Server started, listening on 59365
[info] Gradle client connected to server
[error] [gradle-server] Gradle version is required
[error] Error getting build for c:\Users\admin\projects\demo-gradle: Gradle version is required
[info] Found 0 tasks

```

**Comment by Eskibear** [comment]
FYI, my settings:
![image](https://user-images.githubusercontent.com/2351748/129998568-5cc74ba8-7674-4566-9e8f-e249d39ea845.png)

**Comment by Eskibear** [comment]
after changing to use gradle wrapper, tasks are correctly listed, but dependencies are still not there, and the status bar error message is the same. 

logs from output: 
[log.txt](https://github.com/microsoft/vscode-gradle/files/7011458/log.txt)

![image](https://user-images.githubusercontent.com/2351748/130001783-ff9760bf-2777-4774-adc1-d13d1908c65c.png)

**Comment by CsCherrYY** [comment]
> after changing to use gradle wrapper, tasks are correctly listed, but dependencies are still not there, and the status bar error message is the same.
> 
> logs from output:
> [log.txt](https://github.com/microsoft/vscode-gradle/files/7011458/log.txt)
> 
> ![image](https://user-images.githubusercontent.com/2351748/130001783-ff9760bf-2777-4774-adc1-d13d1908c65c.png)

This comes from circular dependencies and has been fixed in #923. For the Gradle local problem, I'll continue to investigate and fix

related issue: #819

**Comment by CsCherrYY** [comment]
Here are some findings about this issue:

related discusstion: https://github.com/microsoft/vscode-gradle/issues/243#issuecomment-621174072


In vscode-java, there are a lot of Gradle related settings, they are:
- `java.import.gradle.home`: The local gradle distribution path. See https://docs.gradle.org/current/javadoc/org/gradle/api/invocation/Gradle.html#getGradleHomeDir--
- `java.import.gradle.user.home`: This directory is used to cache downloaded resources, compiled build scripts and so on. See https://docs.gradle.org/current/javadoc/org/gradle/api/invocation/Gradle.html#getGradleUserHomeDir--
- `java.import.gradle.version`:  The gradle version will be used. If the wrapper is disabled or missing
- `java.import.gradle.wrapper.enabled`: Use wrapper or not

In a short word, `java.import.gradle.user.home` is used to cache resources, and the other three settings are used to specific gradle distribution (**wrapper**, **specific version** or **local distribution**). In the reference: https://github.com/redhat-developer/vscode-java/blob/master/README.md#supported-vs-code-settings, the priority of the three settings is `java.import.gradle.wrapper.enabled`, `java.import.gradle.version`, `java.import.gradle.home`. If a more front setting is valid (has been set and with a valid value), the rest settings will be ignored. 

Currently, `vscode-gradle` consumes the first two settings, `java.import.gradle.wrapper.enabled`, and `java.import.gradle.version`, with the correct priority. It doesn't support `java.import.gradle.home` currently, that the root cause of this issue. 

So I plan to support `java.import.gradle.home` with a lowest priority among these three settings to support the local distribution scenario. Could you please provide any comments? Or is there any misunderstanding from me in the content? @Eskibear @testforstephen @jdneo

**Comment by CsCherrYY** [comment]
fixed in #926
